lib/topic_query.rb

```diff
@@ -637,9 +637,13 @@ def default_results(options = {})
     options.reverse_merge!(@options)
     options.reverse_merge!(per_page: per_page_setting) unless options[:limit] == false
 
-    # Whether to return visible topics
-    options[:visible] = true if @user.nil? || @user.regular?
-    options[:visible] = false if @user && @user.id == options[:filtered_to_user]
+    # Whether to include unlisted (visible = false) topics
+    viewing_own_topics = @user && @user.id == options[:filtered_to_user]
+
+    if options[:visible].nil?
```

Comment:
We need to prevent overwriting any explicitly passed option. This is especially relevant for flows that are run as system user.